src/workloads/sharding/ReshardCollectionMixed.yml

```diff
@@ -0,0 +1,186 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/sharding"
+Description: |
+  Runs the reshardCollection command while read and write operations are active on the collection
+  being resharded.
+
+  The workload consists of 5 phases:
+    1. Creating an empty sharded collection distributed across all shards in the cluster.
+    2. Populating the sharded collection with data.
+    3. Running read and write operations on the collection before it is resharded.
+    4. Running read and write operations on the collection while it is being resharded.
+    5. Running read and write operations on the collection after it has been resharded.
+
+  The inserted documents have the following form:
+
+      {_id: 10, oldKey: 20, newKey: 30, counter: 0, padding: 'random string of bytes ...'}
+
+  The collection is initially sharded on {oldKey: 'hashed'} and then resharded on {newKey: 1}.
+
+GlobalDefaults:
+  Nop: &Nop {Nop: true}
+
+  Database: &Database test
+  # Collection0 is the default collection populated by the MonotonicSingleLoader.
+  Collection: &Collection Collection0
+  Namespace: &Namespace test.Collection0
+
+  # Note that the exact document size may exceed ApproxDocumentSize because of field names and other
+  # fields in the document.
+  ApproxDocumentSize: &ApproxDocumentSize 2000  # = 2kB
+  ApproxDocumentSize50Pct: &ApproxDocumentSize50Pct 1000  # = 1kB
+  DocumentCount: &DocumentCount 2_000_000  # for an approximate total of 3GB
+  NumChunks: &NumChunks 60                 # with approximately 50MB per chunk
+
+  ShardKeyValueMin: &ShardKeyValueMin 1
+  # We use 100x the number of chunks so that the $sample aggregation to choose the new split points
+  # has a wide range of distinct values to choose from. It could be even larger.
+  ShardKeyValueMax: &ShardKeyValueMax 6000
+
+  ReadOperations: &ReadOperations
+  - OperationName: findOne
+    OperationCommand:
+      Filter: {_id: {^RandomInt: {min: 1, max: *DocumentCount}}}
+
+  WriteOperations: &WriteOperations
+  - OperationName: updateOne
+    OperationCommand:
+      Filter: {_id: {^RandomInt: {min: 1, max: *DocumentCount}}}
+      Update: {$inc: {counter: 1}}
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 201
+  # The reshardCollection command is expected to take a long to complete so we give the actor
+  # running it a much higher socket timeout.
+  ReshardCollection:
+    QueryOptions:
+      maxPoolSize: 1
+      socketTimeoutMS: 36_000_000  # = 10 hours
```

Comment:
Don't we have a 6 hour timeout on DSI instance? Just checking if this value is actually honored, or we should use a smaller value.

Reply:
I used 10 hours as a round "hand" number than doing the math for 6 hours. The sys-perf task would fail if it took longer than 6 hours still. The resharding operation after d1eb8af5673d1a6130805e435791c24ed8ce9580 and 2044fe6532104a0eb63f94c3afb2393108884f02 finished in 26 minutes. I don't actually need to raise the socket timeout this much anymore but feel like keeping it as-is.